pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -2556,3 +2550,57 @@ func (a *ApplicationBackupController) validateApplicationBackupParameters(backup
 	}
 	return nil
 }
+
+// handleCSINametoCSIMapMigration migrates the CSISnapshotName variable to CSISnapshotMap if "Default" is given
+func handleCSINametoCSIMapMigration(spec stork_api.ApplicationBackupSpec) error {
+	// Check whether if VolumeSnapshotClassName is given. If yes, check it's using the older way of requestParams. If yes, then migrate
+	// to new map in case of csi based backups
+	if spec.Options != nil {
+		if snapshotClassName, ok := spec.Options[optCSISnapshotClassName]; ok && len(spec.CSISnapshotClassMap) == 0 {
+			vsc, err := externalsnapshotter.Instance().GetSnapshotClass(snapshotClassName)
+			// In Case of Default given, list out all csi drivers and their snapshot classes and make a map of default vsc to its respective csi driver
+			// make empty map if no default vsc found
+			if k8s_errors.IsNotFound(err) && strings.EqualFold(snapshotClassName, "default") {
+				spec.CSISnapshotClassMap = make(map[string]string)
+
+				// list all csi drivers in the cluster
+				drivers, dErr := storage.Instance().ListCsiDrivers()
+				if dErr != nil {
+					return dErr
+				}
+
+				// list all snapshot classes in the cluster
+				snapshotClasses, sErr := externalsnapshotter.Instance().ListSnapshotClasses()
+				if sErr != nil {
+					return sErr
+				}
+
+				for _, driver := range drivers.Items {
+					spec.CSISnapshotClassMap[driver.Name] = ""
+					driverCount := 0
+					for _, vsc := range snapshotClasses.Items {
+						if vsc.Driver == driver.Name {
+							driverCount++
+							if isTrue, exist := vsc.GetAnnotations()[defaultVolumeSnapshotClassAnnotation]; isTrue == "true" && exist {
+								spec.CSISnapshotClassMap[driver.Name] = vsc.Name
+								break
+							} else if driverCount == 1 {
+								spec.CSISnapshotClassMap[driver.Name] = vsc.Name
+							} else if driverCount > 1 {
```

Comment:
If we have 5 snapshotClasses matching the driverName we will end up resetting it to + "" all 5 times. Not required.